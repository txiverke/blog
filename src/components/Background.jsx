// @flow

import React from 'react'

import landingBg from '../assets/imgs/bg_1.png'

class Background extends React.PureComponent {
  state = {
    render: ''
  }
  
  props: {
    title: string,
  }

  componentDidMount() {
    setTimeout(() => { 
      this.setState({ render: 'app-background-render' }) 
    })
  }

  render() {
    const render = this.state.render
    
    return (
      <figure className={`app-background ${render}`}>
        <img 
          className="app-background-img" 
          src={landingBg} 
          alt={`Background of ${this.props.title}`} 
        />
      </figure>
    )
  }
}

export default Background