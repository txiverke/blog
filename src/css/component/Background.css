.app-background {
  position: absolute;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  width: 100%;
  height: 100vh;
  overflow: hidden;
  opacity: 0;
  transition-property: opacity;
  transition-duration:.5s;
  transition-timing-function: linear;
  
}

.app-background-render {
  opacity: 1
}

@media only screen 
  and (min-width: 850px) {
    .app-background img {
      background-size: cover !important;
      position: relative;
      width: 100%;
      height: auto;
      top: 50%;
      transform: translateY(-50%)
    }
  }

@media only screen 
  and (max-width: 849px) {
    .app-background img {
      background-size: cover !important;
      position: relative;
      width: auto;
      height: 100vh;
      left: 50%;
      transform: translateX(-50%)
    }
}

@media only screen  
  and (max-device-width: 849px) 
  and (orientation: landscape) { 
    .app-background img {
      background-size: cover !important;
      position: relative;
      width: 100%;
      height: auto;
      top: -25%;
      left: 50%;
      transform: translateX(-50%);
    }
}
