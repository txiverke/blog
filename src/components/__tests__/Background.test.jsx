// @flow

import React from 'react'
import { shallow } from 'enzyme'
import Background from '../Background'

jest.useFakeTimers()

describe('<Background />', () => {
  it('should render itself', () => {
    const title = 'Just a fake title'
    const component = shallow(<Background title={title} />)
    expect(component).toMatchSnapshot()
    expect(component.find('img').props().alt).toEqual(`Background of ${title}`)
    expect(setTimeout).toHaveBeenCalledTimes(1);
  })
})