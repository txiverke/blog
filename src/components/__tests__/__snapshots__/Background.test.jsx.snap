// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Background
    title="Just a fake title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="Background of Just a fake title"
        className="app-background-img"
        src="bg_1.jpg"
/>,
      "className": "app-background ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Background of Just a fake title",
        "className": "app-background-img",
        "src": "bg_1.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "figure",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Background of Just a fake title"
          className="app-background-img"
          src="bg_1.jpg"
/>,
        "className": "app-background ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Background of Just a fake title",
          "className": "app-background-img",
          "src": "bg_1.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "figure",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
